pkg/gather/clusterconfig/operators.go

```diff
@@ -0,0 +1,368 @@
+		}
+	}
+	return res
+}
+
+func getOperatorResourcesVersions(discoveryClient discovery.DiscoveryInterface) (map[string][]string, error) {
+	resources, err := discoveryClient.ServerPreferredResources()
+	if err != nil {
+		return nil, err
+	}
+	resourceVersionMap := make(map[string][]string)
+	for _, v := range resources {
+		if strings.Contains(v.GroupVersion, "operator.openshift.io") {
+			gv, err := schema.ParseGroupVersion(v.GroupVersion)
+			if err != nil {
+				continue
+			}
+			for _, ar := range v.APIResources {
+				key := fmt.Sprintf("%s-%s", gv.Group, ar.Name)
+				r, ok := resourceVersionMap[key]
+				if !ok {
+					resourceVersionMap[key] = []string{gv.Version}
+				} else {
+					r = append(r, gv.Version)
+				}
+			}
+		}
+	}
+	return resourceVersionMap, nil
+}
+
+// ClusterOperatorAnonymizer implements serialization of ClusterOperator without change
+type ClusterOperatorAnonymizer struct{ *configv1.ClusterOperator }
+
+// Marshal serializes ClusterOperator
+func (a ClusterOperatorAnonymizer) Marshal(_ context.Context) ([]byte, error) {
+	return runtime.Encode(openshiftSerializer, a.ClusterOperator)
+}
+
+// GetExtension returns extension for anonymized cluster operator objects
+func (a ClusterOperatorAnonymizer) GetExtension() string {
+	return "json"
+}
+
+func isHealthyOperator(operator *configv1.ClusterOperator) bool {
+	for _, condition := range operator.Status.Conditions {
+		switch {
+		case condition.Type == configv1.OperatorDegraded && condition.Status == configv1.ConditionTrue,
+			condition.Type == configv1.OperatorAvailable && condition.Status == configv1.ConditionFalse:
+			return false
+		}
+	}
+	return true
+}
+
+func namespacesForOperator(operator *configv1.ClusterOperator) []string {
+	var ns []string
+	for _, ref := range operator.Status.RelatedObjects {
+		if ref.Resource == "namespaces" {
+			ns = append(ns, ref.Name)
+		}
+	}
+	return ns
+}
+
+// ClusterOperatorResourceAnonymizer implements serialization of clusterOperatorResource
+type ClusterOperatorResourceAnonymizer struct{ resource clusterOperatorResource }
+
+// Marshal serializes clusterOperatorResource with IP address anonymization
+func (a ClusterOperatorResourceAnonymizer) Marshal(_ context.Context) ([]byte, error) {
+	bytes, err := json.Marshal(a.resource)
+	if err != nil {
+		return nil, err
+	}
+	resStr := string(bytes)
+	//anonymize URLs
+	re := regexp.MustCompile(`"(https|http)://(.*?)"`)
+	urlMatches := re.FindAllString(resStr, -1)
+	for _, m := range urlMatches {
+		m = strings.ReplaceAll(m, "\"", "")
+		resStr = strings.ReplaceAll(resStr, m, anonymizeString(m))
+	}
+	// anonymize IP addresses
+	re = regexp.MustCompile(`(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}`)
+	ipMatches := re.FindAllString(resStr, -1)
+	for _, m := range ipMatches {
+		resStr = strings.ReplaceAll(resStr, m, anonymizeString(m))
+	}
```

Comment:
I guess we can remove the IP address anonymization. I think I removed it in master.